ThunderCloud/QuizGridItem.swift

```diff
@@ -0,0 +1,33 @@
+//
+//  QuizGridItem.swift
+//  ThunderCloud
+//
+//  Created by Simon Mitchell on 16/02/2018.
+//  Copyright © 2018 threesidedcube. All rights reserved.
+//
+
+import UIKit
+import ThunderCollection
+
+/// Used to display a quiz badge in a collection view
+open class QuizGridItem: GridItem {
+	
+	let badgeId: String?
+
+	public required init?(dictionary: [AnyHashable : Any]) {
+		
+		if let badgeId = dictionary["badgeId"] as? String {
+			self.badgeId = badgeId
+		} else if let badgeId = dictionary["badgeId"] as? Int {
+			self.badgeId = "\(badgeId)"
+		} else {
+			badgeId = nil
```

Comment:
`badgeId` is nil by default so this else is not required

Reply:
It is, because `badgeId` is a `let` 😝

Reply:
Would suggest a private var again here with a default assignment of nil

Reply:
I believe badgeId is used outside of the file here, so will try just making it internal for use in Storm 👍

Reply:
Actually I'm not sure I agree with this... I think badgeId shouldn't be made variable, even internally for the sake of removing a line of code. I really feel like if something does and should not change then it should be reflected in it's API, whether that's just internal to the framework or made public to the consumer 😊

Reply:
Never mind, look at me flip-flopping over here. it isn't used even outside this class so it can remain private var 😂